catalyst/dl/experiment/gan.py

```diff
@@ -0,0 +1,158 @@
+from collections import OrderedDict
+from typing import Dict, List, Union
+
+from torch.utils.data import DataLoader
+
+from catalyst.dl import Callback
+from catalyst.dl.callbacks import (
+    CheckpointCallback, ConsoleLogger, PhaseBatchWrapperCallback,
+    PhaseManagerCallback, RaiseExceptionCallback, VerboseLogger
+)
+from catalyst.utils.tools.typing import Criterion, Model, Optimizer, Scheduler
+from .base import BaseExperiment
+
+
+class GanExperiment(BaseExperiment):
+    """
+    One-staged GAN experiment
+    Mainly used in notebooks
+    """
+
+    def __init__(
+        self,
+        model: Union[Model, Dict[str, Model]],
```

Comment:
should it work if we pass only one model instead of a dict?

Reply:
No, dict intended nearly always. removed union with model part.